### PR TITLE
feat: highlight search matches and search result score

### PR DESCRIPTION
**Pull request by willothy**

### CODE REVIEW

File: lua/hawtkeys/ui.lua [hunk 1]
```diff
@@ -138,6 +114,48 @@ M.search = function(text)
...
+                    )
+                    marked[char] = pos -- Mark this character's position
+                end
+            end
+
+            local score_offset = #line - #tostring(data.score)
+            local score_hl = "MoreMsg"
+            if data.score <= 0 then
+                score_hl = "ErrorMsg"
+            elseif data.score <= 2 then
```
Comment (tris203) [inline lua/hawtkeys/ui.lua]: Can we add a if =>6 then a SucessMsg?

File: lua/hawtkeys/ui.lua [hunk 2]
```diff
@@ -80,15 +57,14 @@ M.search = function(text)
...
-        table.insert(
-            lines,
-            "Key: "
-                .. highlight_desc(text, data.combo)
-                .. "<leader>"
-                .. data.combo
-                .. " - Hawt Score: "
-                .. data.score
+        local line = string.format(
+            "Key: %s <leader>%s - Hawt Score: %d",
```
Comment (tris203) [inline lua/hawtkeys/ui.lua:68]: Love that we are using template strings more. Good job

File: lua/hawtkeys/ui.lua (hunk 1 above)
Comment (tris203) [reply lua/hawtkeys/ui.lua]: Also, instead of hardcoding the scores. Can you define a stable at the top for low score, midscore and highscore?

**Review by tris203 — CHANGES_REQUESTED** [review]
just a couple of niggles then LGTM

File: lua/hawtkeys/ui.lua (hunk 1 above)
Comment (willothy) [reply lua/hawtkeys/ui.lua]: > Can we add a if =>6 then a SucessMsg?

MoreMsg is green for me, SuccessMsg is not a default highlight group as far as I know.

**Comment by willothy** [comment]
You should try this out if you haven't to see if you like the way it looks, I'm not sure how it will be with colorschemes other than my own. We could also set our own highlight groups for it and allow configuration of that.

**Comment by tris203** [comment]
> You should try this out if you haven't to see if you like the way it looks, I'm not sure how it will be with colorschemes other than my own. We could also set our own highlight groups for it and allow configuration of that.

yeah, its a bit wierd on my theme. Moremsg is blue.

I would like to move the colours to config, but we also need sensible defaults. What about using the colours for the notify messages?
EDIT: or the healthmessages, there is healtherror, healthsuccess and healthwarning

**Comment by tris203** [comment]
or the git colours
```lua
  git = {
    add = "#4197a4", -- high score
    change = "#506d9c", -- mid score
    delete = "#c47981", -- low score
    ignore = "#8990b3" --already mapped
  },
  ```

File: lua/hawtkeys/ui.lua (hunk 1 above)
Comment (willothy) [reply lua/hawtkeys/ui.lua]: I added an underline with a custom hlgroup for matches >=6

**Comment by willothy** [comment]
> > You should try this out if you haven't to see if you like the way it looks, I'm not sure how it will be with colorschemes other than my own. We could also set our own highlight groups for it and allow configuration of that.
> 
> yeah, its a bit wierd on my theme. Moremsg is blue.
> 
> I would like to move the colours to config, but we also need sensible defaults. What about using the colours for the notify messages? EDIT: or the healthmessages, there is healtherror, healthsuccess and healthwarning

I think health messages is the move. Those should usually be the same colors, but I've seen a lot of variation in git colors

Edit: are health messages default highlights? They're not defined for me even on default colorscheme.

**Comment by tris203** [comment]
```
f fg a m 0 p
```
is a search that should bring back all score options

i will test this when i get home

**Review by tris203 — APPROVED** [review]

**Comment by tris203** [comment]
another great contribution, thanks!

**Pull request merged**